src/main/java/de/hysky/skyblocker/skyblock/dwarven/CrystalsLocationsManager.java

```diff
@@ -131,20 +178,59 @@ private static void registerWaypointLocationCommands(CommandDispatcher<FabricCli
-        Color locationColor = WAYPOINT_LOCATIONS.get(location).color;
-        text.append(Text.literal(location).withColor(locationColor.getRGB()));
-        text.append(Text.literal(" at : " + blockPos.getX() + " " + blockPos.getY() + " " + blockPos.getZ() + "."));
+        text.append(Text.translatable("skyblocker.config.mining.crystalsWaypoints.addedWaypoint"));
+        int locationColor = WAYPOINT_LOCATIONS.get(location).getColor();
+        text.append(Text.literal(location).withColor(locationColor));
+        text.append(Text.literal(" ").append(Text.translatable("skyblocker.config.mining.crystalsWaypoints.addedWaypoint.at")));
+        text.append(Text.literal(" : " + blockPos.getX() + " " + blockPos.getY() + " " + blockPos.getZ() + "."));
 
         return text;
     }
 
-    private static Text getLocationInputText(String location) {
+    /**
+     * Creates a formated text with a list of possible places to add a waypoint for
+     *
+     * @param location       the location where the waypoint will be created
+     * @param excludeUnknown if the "Unknown" location should be available to add
+     * @return text for a message to send to the player
+     */
+    private static Text getLocationMenu(String location, boolean excludeUnknown) {
         MutableText text = Constants.PREFIX.get();
 
+        //if the user has all available waypoints active warn them instead of an empty list (excused unknown from check when disabled)
+        if (activeWaypoints.size() == WAYPOINT_LOCATIONS.size() || (excludeUnknown && WAYPOINT_LOCATIONS.size() - activeWaypoints.size() == 1 && !activeWaypoints.containsKey(MiningLocationLabel.CrystalHollowsLocationsCategory.UNKNOWN.getName()))) {
+            return text.append(Text.translatable("skyblocker.config.mining.crystalsWaypoints.allActive").formatted(Formatting.RED));
+        }
+
         for (String waypointLocation : WAYPOINT_LOCATIONS.keySet()) {
-            Color locationColor = WAYPOINT_LOCATIONS.get(waypointLocation).color;
-            text.append(Text.literal("[" + waypointLocation + "]").withColor(locationColor.getRGB()).styled(style -> style.withClickEvent(new ClickEvent(ClickEvent.Action.RUN_COMMAND, "/skyblocker crystalWaypoints " + location + " " + waypointLocation))));
+            //do not show option to add waypoints for existing locations or unknown if its disabled
+            if (activeWaypoints.containsKey(waypointLocation) || (excludeUnknown && Objects.equals(waypointLocation, MiningLocationLabel.CrystalHollowsLocationsCategory.UNKNOWN.getName()))) {
+                continue;
+            }
+            int locationColor = WAYPOINT_LOCATIONS.get(waypointLocation).getColor();
+            text.append(Text.literal("[" + waypointLocation + "]").withColor(locationColor).styled(style -> style.withClickEvent(new ClickEvent(ClickEvent.Action.RUN_COMMAND, "/skyblocker crystalWaypoints add " + location + " " + waypointLocation))));
```

Comment:
I think it's a bit difficult for a new player to understand how this feature is supposed to work and that it could be expressed more clearly. 

Some suggestions are:
- Append something like `Mark this location as: ` to the prefix constant and turn the message into `[Skyblocker] Mark this location as: [etc]`. Could be the x y z coords instead of the word `location` too.
- Add hover events with a message indicating clickability, as you expect something clickable to react in some way when you hover over it.